src/librustc_codegen_llvm/declare.rs

```diff
@@ -92,4 +93,19 @@ impl DeclareMethods<'tcx> for CodegenCx<'ll, 'tcx> {
             if !declaration { Some(val) } else { None }
         })
     }
+
+    fn define_sir_type(&self, ty: ykpack::Ty) -> ykpack::TypeId {
+        let mut types = self.sir.as_ref().unwrap().types.borrow_mut();
+        (types.crate_hash, types.index(ty))
```

Comment:
Perhaps I was expecting type IDs to be per-cgu too? No?

Reply:
The now admittedly confusingly names `crate_hash` in `types` is different per-cgu: https://github.com/softdevteam/ykrustc/pull/125/files#diff-ac6d225f5a8ce5d7dfbd7cdc90878877R197-R199

Reply:
I also wonder if we can get these into a yorick-specfic file, to minimise merge conflicts?

Reply:
I could add a new trait.

Reply:
> The now confusingly named crate_hash in types is different per-cgu

Yes, let's rename that please.

Reply:
> I could add a new trait.

Ah, I wouldn't go that far. Perhaps leave them as is then.

Reply:
I'm struggling to find the code where the crate hash in the type id is changed to incorporate the crate disambiguator (the link above doesn't seem to work any more). I can see where the section name (and it's symbol) are made unique, but not in the type ID. Can you point it out please?

Reply:
That is https://github.com/softdevteam/ykrustc/pull/125/files#diff-ac6d225f5a8ce5d7dfbd7cdc90878877R199

Reply:
OK, got it:
```
  198         tcx.crate_hash(LOCAL_CRATE).hash(&mut crate_hash_hasher);                                  
  199         cgu_name.hash(&mut crate_hash_hasher);
```

So crate hash + cgu name = our new hash. LGTM.